### PR TITLE
PAYARA-3386: Update login.conf with login module

### PR DESCRIPTION
**Pull request by pdudits**
Login Module registration can be requested via `--login-module` parameter to `create-auth-realm`. In such case property `jaas-context` is required for the realm.

In the Admin UI, the default jaas context is now prefilled when creating a predefined realm. Login modules can be registered by enabling a checkbox and filling in the login module class name (and JAAS context if not provided by predefined class)

### CODE REVIEW

**Comment by arjantijms** [comment]
Jenkins test please

File: appserver/admingui/common/src/main/java/org/glassfish/admingui/common/handlers/SecurityHandler.java [hunk 1]
```diff
@@ -82,7 +82,12 @@ public static void getRealmAttrForCreate(HandlerContext handlerCtx) {
         handlerCtx.setOutputValue("realmClasses", realmClassList);
         handlerCtx.setOutputValue("classnameOption", "predefine");
         Map attrMap = new HashMap();
+        attrMap.put("fileJaax", "fileRealm");
+        attrMap.put("ldapJaax", "ldapRealm" );
+        attrMap.put("solarisJaax", "solarisRealm");
+        attrMap.put("jdbcJaax", "jdbcRealm");
```
Comment (pdudits) [inline appserver/admingui/common/src/main/java/org/glassfish/admingui/common/handlers/SecurityHandler.java]: image

File: appserver/admingui/common/src/main/resources/security/realms/realmAttrs.inc [hunk 2]
```diff
@@ -194,3 +194,18 @@
...
+        <sun:property id="jaax" labelAlign="left" noWrap="#{true}" overlapLabel="#{false}" label="$resource{i18nc.realm.jaax}" helpText="$resource{i18nc.realm.jaaxHelp} $resource{i18nc.realm.loginmodule.jaaxNote}" >
+            <!afterCreate
+                getClientId(component="$this{component}" clientId=>$page{loginModuleJaax});
+            />
+            <sun:textField id="jaax" columns="$int{60}" maxLength="#{sessionScope.fieldLengths['maxLength.realm.jaax']}" text="#{attrMap.loginModuleJaax}"/>
+        </sun:property>
+        <sun:property id="loginModule" labelAlign="left" noWrap="#{true}" overlapLabel="#{false}" label="$resource{i18nc.realm.loginmodule.class}" helpText="$resource{i18nc.realm.loginmodule.classHelp}" >
+            <sun:textField id="loginModule" text="#{attrMap.loginModuleClass}" columns="$int{57}" maxLength="#{sessionScope.fieldLengths['maxLength.realm.Classname']}" />
+        </sun:property>
+    </sun:propertySheetSection>
```
Comment (pdudits) [inline appserver/admingui/common/src/main/resources/security/realms/realmAttrs.inc]: image

File: appserver/admingui/common/src/main/resources/security/realms/realmjs.inc [hunk 3]
```diff
@@ -118,8 +118,12 @@ function showDisplay(testtype) {
     document.getElementById('#{certSectionId}').style.display = 'none'; 
     document.getElementById('#{solarisSectionId}').style.display = 'none';
     document.getElementById('#{pamSectionId}').style.display = 'none';
-    
-    if (testtype.match('none')){  
+    var loginModuleJaax = document.getElementById('#{loginModuleJaax}');
+
+    loginModuleJaax.style.display = 'none';
```
Comment (pdudits) [inline appserver/admingui/common/src/main/resources/security/realms/realmjs.inc]: For predefined classes, that already have an attribute for JAAS Context, it is not displayed for login module configuration:

image

**Comment by pdudits** [comment]
jenkins test please

**Comment by pdudits** [comment]
jenkins test please

**Comment by Cousjava** [comment]
Further to my review: the issue is that I did not supply all fields and the UI did not indicate that I needed to or throw an error when I submitted it

**Comment by pdudits** [comment]
@Cousjava, I improved the UI a bit, try again, please

**Comment by pdudits** [comment]
jenkins test

**Pull request merged**